scipy/sparse/_construct.py

```diff
@@ -20,10 +20,12 @@
 from ._coo import coo_matrix
 from ._dia import dia_matrix
 
+from ._arrays import coo_array, csr_array, csc_array, dia_array
+
 from ._base import issparse
 
 
-def spdiags(data, diags, m, n, format=None):
+def spdiags(data, diags, m, n, format=None, _array=False):
```

Comment:
I'm not sure how I feel about adding semi-private kwargs to these sparse._construct functions. I would prefer using some tricks with the `format=` kwarg instead, like allowing `format='coo_array'`.

Reply:
Perhaps another option here would be a keyword-only argument? Seems like a more convenient interface and more transparent for backwards compatibility at a later stage.

Reply:
This is currently only used to provide `eye` on sparse arrays for `__pow__`, which handles the power zero special case as follows:

```
            if other == 0:
                from ._construct import eye
                return eye(
                    self.shape[0], dtype=self.dtype, _array=self._is_array
                )
```

I can definitely get by without it.

Reply:
OK, I've removed these.